### PR TITLE
Nfc writer

### PR DESCRIPTION
**Pull request by GharibGharib**
### Changes / Veranderingen
Interface Class toegevoegd.

### Checklist

- [x] The code is tested / De code is getest

### Issue

**Issue:** De code werkt in de praktijk. En vraag om een ​​code review#

### CODE REVIEW

File: Source/NFC-Writer/src/MFRC522Writer.cpp [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    if (!GetCardAvailable())
+    {
+        return -1;
+    }
+    if (message == nullptr)
+    {
+        return -1;
+    }
+
+    byte blockAddr4 = 4;
```
Comment (NathanThus) [inline Source/NFC-Writer/src/MFRC522Writer.cpp:50]: No. So much no.

File: Source/NFC-Writer/src/MFRC522Writer.cpp [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    byte blockAddr5 = 5;
+    char inputString[maxMessageLength] = "";
```
Comment (NathanThus) [inline Source/NFC-Writer/src/MFRC522Writer.cpp:52]: Why.

File: Source/NFC-Writer/src/UI.cpp [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+int UI::ReadTextFromSerial(char *message, int max_message_length)
+{
+    Serial.print("The max text is: ");
+    Serial.print(max_message_length);
+    Serial.println(" Characters");
+    unsigned int message_pos = 0;
+    message[0] = '\0';
+    char messagetemp[1000] = "";
```
Comment (NathanThus) [inline Source/NFC-Writer/src/UI.cpp:14]: WHY.

**Review by TomVer99 — CHANGES_REQUESTED** [review]
PR heeft 0 info, vul dit in.

File: Source/NFC-Writer/src/UI.cpp [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+UI::UI()
+{
+}
+
+int UI::ReadTextFromSerial(char *message, int max_message_length)
+{
+    Serial.print("The max text is: ");
+    Serial.print(max_message_length);
+    Serial.println(" Characters");
+    unsigned int message_pos = 0;
```
Comment (NathanThus) [inline Source/NFC-Writer/src/UI.cpp:12]: ???

**Review by NathanThus — COMMENTED** [review]
Lot's of magic numbers.

**Review by NathanThus — CHANGES_REQUESTED** [review]
Making these changes mandatory.

File: Source/NFC-Writer/src/MFRC522Writer.cpp [hunk 5]
```diff
@@ -0,0 +1,155 @@
+#include "MFRC522Writer.hpp"
+
+MFRC522Writer::MFRC522Writer(MFRC522 &nfc)
+{
+    mfrc522 = nfc;
```
Comment (TomVer99) [inline Source/NFC-Writer/src/MFRC522Writer.cpp:5]: Gebruik init list.

File: Source/NFC-Writer/src/MFRC522Writer.cpp [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+        {
+            dataBuffer2[stringLength2 + j] = 0;
+        }
+    }
+
+    if (AuthForWrite() == -1)
+    {
+        return -1;
+    }
+    if (Transfer(dataBuffer, blockAddr4) == -1)
```
Comment (TomVer99) [inline Source/NFC-Writer/src/MFRC522Writer.cpp:101]: Kunnen deze if statements samen?

File: Source/NFC-Writer/src/MFRC522Writer.cpp [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+    {
+        mfrc522.PICC_IsNewCardPresent();
+        return mfrc522.PICC_ReadCardSerial();
+    }
+    return true;
+}
+
+int MFRC522Writer::Write(char *message)
+{
+    if (!GetCardAvailable())
```
Comment (TomVer99) [inline Source/NFC-Writer/src/MFRC522Writer.cpp:41]: If statements kunnen samen?

**Review by TomVer99 — CHANGES_REQUESTED** [review]
Have you implemented any feedback from the last time?

File: Source/NFC-Writer/src/MFRC522Writer.cpp (hunk 2 above)
Comment (NathanThus) [reply Source/NFC-Writer/src/MFRC522Writer.cpp:49]: Why do you need two input strings?

File: Source/NFC-Writer/src/MFRC522Writer.cpp [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+    {
+        return -1;
+    }
+    int stringLength2 = strlen(inputString2);
+    if (stringLength2 > bufferSizeBlock5)
+    {
+        return -1;
+    }
+    byte dataBufferBlock4[bufferSizeBlock4] = "";
+    byte dataBufferBlock5[bufferSizeBlock5] = "";
```
Comment (NathanThus) [inline Source/NFC-Writer/src/MFRC522Writer.cpp:78]: Why do you need so many blocks?

File: Source/NFC-Writer/src/UATRInterface.cpp [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+UARTI::~UARTI()
+{
+    delete (inputBuffer);
+    inputBuffer = nullptr;
+}
+
+int UARTI::ReadTextFromSerial(char* message, int max_message_length)
+{
+    bool printMessage = false;
+    message[0] = '\0';
```
Comment (NathanThus) [inline Source/NFC-Writer/src/UATRInterface.cpp:14]: Why are you immediately setting [0] to '\0'

**Review by NathanThus — CHANGES_REQUESTED** [review]

**Review by NathanThus — APPROVED** [review]

**Comment by NathanThus** [comment]
Decent enough. I'll allow it through due to time crunch, but keep in mind what I said about code styling @GharibGharib

**Pull request merged**